### PR TITLE
[Yaml] support to parse and dump DateTime objects

### PR DESCRIPTION
**Pull request by xabbuh**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #6275, #8580, #11838, #14420 |
| License | MIT |
| Doc PR | symfony/symfony-docs#6226 |

### CODE REVIEW

File: src/Symfony/Component/Yaml/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,14 @@ CHANGELOG
 3.1.0
 -----
 
+ * Added support for parsing timestamps as `\DateTime` objects:
+
+   ```php
+   Yaml::parse('2001-12-15 21:59:43.10 -5', Yaml::Parse_DATETIME);
```
Comment (mpajunen) [inline src/Symfony/Component/Yaml/CHANGELOG.md]: Should be capitalized: `Yaml::PARSE_DATETIME`.

Comment (xabbuh) [inline src/Symfony/Component/Yaml/CHANGELOG.md]: Good catch. :) Thanks!

File: src/Symfony/Component/Yaml/Inline.php [hunk 2]
```diff
@@ -568,8 +574,15 @@ private static function evaluateScalar($scalar, $references = array())
                         return log(0);
                     case preg_match('/^(-|\+)?[0-9,]+(\.[0-9]+)?$/', $scalar):
                         return (float) str_replace(',', '', $scalar);
+                    case preg_match(self::getTimestampRegex(), $scalar) && Yaml::PARSE_DATETIME & $flags:
+                        return new \DateTime($scalar,new \DateTimeZone('UTC'));
```
Comment (fabpot) [inline src/Symfony/Component/Yaml/Inline.php]: Can we move this into the next `case` to avoid matching twice?

Comment (xabbuh) [inline src/Symfony/Component/Yaml/Inline.php]: Sure, makes sense. Did that.

**Comment by stof** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**